src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -105,7 +107,32 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   private async startConnecting(reconnecting: boolean): Promise<void> {
-    const signedUrl = await this.prepareWebSocketUrl();
+    // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired
+    let endpointUrl = !reconnecting ? this.configuration.endpointUrl : undefined;
+    if (endpointUrl === undefined) {
+      try {
+        if (this.configuration.chimeClient.getMessagingSessionEndpoint instanceof Function) {
```

Comment:
If the `chimeClient` is v3 version then, `this.configuration.chimeClient.getMessagingSessionEndpoint` would error right since `getMessagingSessionEndpoint` may not be in `chimeClient`? So do we have to do a validity check here?

Reply:
Not follow... on v3 client getMessagingSessionEndpoint will be null so fail the instanceof Function check and use the GetMessagingSessionCommand.

Reply:
Cool just wanted to check whether you have already checked `getMessagingSessionEndpoint` validity in here.

Reply:
Isn't that why we check whether `getMessagingSessionEndpoint` is a function?

Reply:
Yup It should be fine I resolved this :)